docs/developer-guides/merkle-root-construction.md

```diff
@@ -2,47 +2,46 @@
-
-If a block only has a coinbase transaction, the coinbase TXID is used as
-the merkle root hash.
-
-If a block only has a coinbase transaction and one other transaction,
-the TXIDs of those two transactions are placed in order, concatenated as
-64 raw bytes, and then SHA256(SHA256()) hashed together to form the
-merkle root.
-
-If a block has three or more transactions, intermediate merkle tree rows
-are formed. The TXIDs are placed in order and paired, starting with the
-coinbase transaction's TXID. Each pair is concatenated together as 64
-raw bytes and SHA256(SHA256()) hashed to form a second row of
-hashes. If there are an odd (non-even) number of TXIDs, the last TXID is
-concatenated with a copy of itself and hashed. If there are more than
-two hashes in the second row, the process is repeated to create a third
-row (and, if necessary, repeated further to create additional rows).
-Once a row is obtained with only two hashes, those hashes are concatenated and
-hashed to produce the merkle root.
-
-<!-- built block 170's merkle root with Python to confirm left-to-right order
-     for A|B concatenation demonstrated below:
-         sha256(sha256("82501c1178fa0b222c1f3d474ec726b832013f0a532b44bb620cce8624a5feb1169e1e83e930853391bc6f35f605c6754cfead57cf8387639d3b4096c54f18f4".decode("hex")).digest()).digest().encode("hex_codec")
--->
-
-![Example Merkle Tree Construction](/img/dev/en-merkle-tree-construction.svg)
-
-TXIDs and intermediate hashes are always in internal byte order when they're
-concatenated, and the resulting merkle root is also in internal byte
-order when it's placed in the block header.
+This page explains what merkle trees are, how Decred uses them and provides step-by-step instructions for constructing compliant Merkle trees.
+
+### Merkle Trees
+
+
+A merkle tree is a data structure that can be used to store hashes of arbitrary data. In the context of Decred, this might be transaction or ticket data. 
+
+Merkle trees are created by repeatedly hashing pairs of nodes until there is only one hash left, the Merkle Root. They are constructed from the bottom up, from hashes of the leaf nodes (hashes of the raw input data). The diagram below shows a merkle root constructed from eight leaf nodes. 
+
+![Merkle Tree](/img/core-blockchain-concepts/merkle_tree.svg)
+
+Merkle trees are particularly useful for blockchains, as they efficiently and compactly summarize larger data sets.
+
+
+### Merkle Trees in Decred
+
+Merkle roots are used to create compact summaries of all transactions in a block. These merkle roots can then be queried to quickly and efficiently verify whether or not a a transaction is included in a block. 
+
+The block header of each block contains the merkle roots of two merkle trees, the `MerkleRoot` tree and `StakeRoot` tree. The `MerkleRoot` tree contains hashes of all regular transaction IDs. The `StakeRoot` tree contains hashes of stake-based transaction IDs (ticket purchases, votes, revocations, etc.). 
```

Comment:
To avoid the repetition we could say either`The header of each block...` or `Each block header...`

Reply:
Nice compression. Used